src/components/data/DataTable.js

```diff
@@ -324,8 +326,8 @@ const DataTable = props => {
                         extraActions: [
                           // settimeout 0 is needed to delay opening the modaals until after the popup menu closes.
                           // Without this, autofocus doesn't work in the modals.
-                          { label: 'Delete Column', onClick: () => setTimeout(() => setDeletingColumn(attributeName), 0) },
-                          { label: 'Clear Column', onClick: () => setTimeout(() => setClearingColumn(attributeName), 0) }
+                          { label: 'Delete Column', disabled: noEdit, tooltip: noEdit ? 'You don\'t have permission to modify this workspace.' : '', onClick: () => setTimeout(() => setDeletingColumn(attributeName), 0) },
+                          { label: 'Clear Column', disabled: noEdit, tooltip: noEdit ? 'You don\'t have permission to modify this workspace.' : '', onClick: () => setTimeout(() => setClearingColumn(attributeName), 0) }
```

Comment:
since `Utils.editWorkspaceError(workspace)` returns the text of an error message, could you use:
```
disabled: !!noEdit, tooltip: noEdit || ''
```
here? Note the `!!` to coerce `noEdit` to a boolean for the `disabled` value.

Reply:
I like that, addressed in https://github.com/DataBiosphere/terra-ui/pull/2982/commits/ac4994e90646b97468481bc72545f60665b69d77